src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -274,9 +275,15 @@ private List<ProfilerEvent> PollSession(ProfilerSession session)
                     }
                 }
             }
+            catch (XEventException)
+            {
+                SendStoppedSessionInfoToListeners(session.XEventSession.Id);
```

Comment:
What sort of situations cause an `XEventException`? Would it be possible/better to treat it as a transient exception and try again instead of stopping the session? If we do want to stop the session, maybe we should do something to notify the user that it was stopped due to an error? Nothing that needs to be changed in this PR necessarily but something to think about for future improvements.

Reply:
An XEventException is thrown when something disrupts the session's ability to get it's target XML, the error message usually says something about the session being configured wrong and checking to make sure it exists. This is definitely a good area to think about improving for future releases.